### PR TITLE
FWF-5886[bugfix] - included validate API in builder API

### PR DESCRIPTION
**Pull request by Ajay-aot**
## 📝 Pull Request Summary

**Description:**
included validate API in builder API

**Related Jira Ticket:**  
https://aottech.atlassian.net/browse/FWF-5689

**DEPENDENCY PR:**  

**Type of Change:**
- [ ] ✨ Feature
- [x] 🐞 Bug Fix
- [ ] ♻️ Refactor
- [ ] 🧹 Code Cleanup
- [ ] 🧪 Test Addition / Update
- [ ] 🔄 SYNC PR (for syncing different repos)
- [ ] 📦 Dependency / Package Updates
- [ ] 📘 Documentation Update
- [ ] 🚀 Deployment / Config Change / Yaml changes

---

## 💻 Frontend Changes 

**Modules/Components Affected:**

**Summary of Frontend Changes:**

**UI/UX Review:**
- [ ] Required
- [ ] Not Applicable

**Screenshots / Screen Recordings (if applicable):**

---

## ⚙️ Backend Changes (Java / Python)

**Modules/Endpoints Affected:**
/form-flow-builder
/form-flow-builder/<int:mapper_id>

**Summary of Backend Changes:**
included validate API in builder API

**API Testing:**
- [x] Done
- [ ] Not Applicable

**Screenshots / Screen Recordings (if applicable):**
Screenshot 2026-01-28 130400

## ✅ Checklist

- [ ] Code builds successfully without lint or type errors  
- [x] Unit tests added or updated [Backend]
- [ ] UI verified [Frontend]   
- [ ] Documentation updated (README / Confluence / API Docs)  
- [x] No sensitive information (keys, passwords, secrets) committed  
- [ ] I have updated the **CHANGELOG** with relevant details  
- [x] I have given a **clear and meaningful PR title and description as per standard format**  
- [ ] I have verified **cross-repo dependencies** (if any)  
- [ ] I have confirmed **backward compatibility** with previous releases  

**Details:**

---

## 👥 Reviewer Notes

___

### **PR Type**
Bug fix, Tests, Enhancement

___

### **Description**
- Add `title` and `path` query validation in builder endpoints

- Invoke `validate_form_name_path_title` before create/update

- Expand unit tests for validation success and failure

- Update user login tests with `VIEW_TASKS` role

___

### Diagram Walkthrough

```mermaid
flowchart LR
  Req["Client Request"] -- "POST/PUT with title/path" --> Extract["Extract title/path"]
  Extract -- "present?" --> Check{"title or path provided?"}
  Check -- "yes" --> Validate["validate_form_name_path_title"]
  Check -- "no" --> Skip["Skip validation"]
  Validate --> Process["create/update form process"]
  Skip --> Process
  Process --> Res["Return response"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>form_process_mapper.py</strong><dd><code>Add title/path query validation to builder API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

forms-flow-api/src/formsflow_api/resources/form_process_mapper.py

<ul><li>Add <code>title</code> and <code>path</code> params to <code>@API.doc</code><br> <li> Extract <code>title</code> and <code>path</code> from <code>request.args</code><br> <li> Call <code>validate_form_name_path_title</code> when provided<br> <li> Mirror changes for both POST and PUT endpoints</ul>

</details>

  </td>
  <td><a href="https://github.com/AOT-Technologies/forms-flow-ai/pull/3151/files#diff-03ec1738e9745151123e5c3c2daa8f4b1d590f4e47694dc8ea53f280c6310e7a">+42/-2</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_form_process_mapper.py</strong><dd><code>Extend builder API tests for validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

forms-flow-api/tests/unit/api/test_form_process_mapper.py

<ul><li>Wrap builder POST calls with FormioService.get_form_search patch<br> <li> Add tests for title/path validation success and duplicate cases<br> <li> Cover title-only, path-only, and no-validation scenarios<br> <li> Add invalid title and path format tests</ul>

</details>

  </td>
  <td><a href="https://github.com/AOT-Technologies/forms-flow-ai/pull/3151/files#diff-316bb9879e42371cfeb03bee32945f5f3b5347125c9cff2c10e9c8136e18e783">+264/-16</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_user.py</strong><dd><code>Add role parameter to user login tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

forms-flow-api/tests/unit/api/test_user.py

- Include `role=VIEW_TASKS` in `get_token` calls

</details>

  </td>
  <td><a href="https://github.com/AOT-Technologies/forms-flow-ai/pull/3151/files#diff-cd5b4c9093ae023b29c55d15525fcd0cea675e201cd2d5e7887c59ec21f6e4ac">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Review by Josephalexantony-aot — APPROVED** [review]

File: forms-flow-api/tests/unit/api/test_form_process_mapper.py [hunk 1]
```diff
@@ -1153,3 +1165,239 @@ def test_form_flow_builder_update_invalid_payload(app, client, session, jwt, moc
...
+    form_data = get_formio_form_request_payload()
+    form_data.update({
+        "title": "Unique Test Form",
+        "path": "uniqueTestForm",
+        "name": "uniqueTestForm",
+    })
+
+    payload = {"formData": form_data}
+
+    with patch(
```
Comment (salabh-aot) [inline forms-flow-api/tests/unit/api/test_form_process_mapper.py]: Same as above comment

**Review by salabh-aot — APPROVED** [review]
